### DISCUSSION TITLE
Auto-restart when file change in developpement mode

### DISCUSSION THREAD

**Discussion by anayrat** [sqlpage/SQLPage#495] [Ideas]
Hello,
A cool feature could be an auto reload mode when a file is changed.
For example, hugo perform an autoreload when it detects a file is changed.
It could be useful in development mode :)

**Comment by lovasoa** [comment]
You don't need to restart SQLPage, it already takes file changes into account immediately!

**Reply by lyderic** [reply]
I guess what the OP means is that he would like the page in the browser to be automatically refreshed each time a change is done on a SQLpage file.

Basically, you have two windows opened side by side: one with your editor, one with your browser. Every time you hit Ctrl-S in your editor (or :w in vim), the browser window is automatically refreshed. This avoids having to Alt-Tab into the browser window, hit Ctrl-R and then re-Alt-Tab back to the editor and so on and so forth.

On hugo, it is called [LiveReload](https://bwaycer.github.io/hugo_tutorial.hugo/extras/livereload/). It is a super neat feature indeed!

From the hugo documentation: "_LiveReload works by injecting JavaScript into the pages Hugo generates, which creates a connection from the browser web socket client to the Hugo web socket server._"

**Reply by lovasoa** [reply]
Oh, I see, sorry! 

Would one of you be interested in making a pull request for this?

**Reply by lyderic** [reply]
Implementing this would mean creating a web socket connection between browser and SQLpage server by injecting javascript into the pages generated by SQLpage.

I have had a look at [how it's done for hugo](https://github.com/gohugoio/hugo/tree/master/livereload). Unfortunately, there is as much golang code as JS code. So it is not a mere matter of copying some JS into a template.

**Reply by anayrat** [reply]
Yes, every time a change a file, I have to refresh it. Maybe it is overkill, as you metionnd, SQLPage already take file change immediately. I just have to refresh my brower.
